pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -412,6 +413,60 @@ func TestPriorityQueue_Delete(t *testing.T) {
 	}
 }
 
+func TestPriorityQueue_Activate(t *testing.T) {
+	var objs []runtime.Object
+	q := NewTestQueueWithObjects(context.Background(), newDefaultQueueSort(), objs)
+	qPodInfo := framework.QueuedPodInfo{
+		PodInfo: highPriNominatedPodInfo,
+	}
+
```

Comment:
If we will not remove the `if pInfo == nil` code snippet as you mentioned, whether you can add a testcase cover this logic.

Reply:
`pInfo` is got from either `unschedulableQ` or `podBackoffQ`. So the following test case should cover the `nil` situation for `pInfo`:
- `test PriorityQueue.activate for pod not in unschedulableQ`, since the `unschedulableQ` and `podBackoffQ` is empty. I could add an `assert` line for them to secure.

And I think it's pretty hard to go inside the check block we are talking about:
```go
...
	if obj, exists, _ := p.podBackoffQ.Get(newQueuedPodInfoForLookup(pod)); !exists {
		...  // return
	} else {
		pInfo = obj.(*framework.QueuedPodInfo) // Here, obj exists AND we need pInfo to be nil
	}
...
```

Any suggestion?

Reply:
yes, after I rechecked the logic, I found it's unreachable as the obj enqueued couldn't be nil. Sorry for the confusion.